spec/v1/envelope.md

```diff
@@ -19,5 +19,48 @@ standard DSSE fields.
     specifying its schema.
 -   `payload` MUST be a base64-encoded JSON [Statement].
 
+## File naming convention
+
+If stored in a dedicated file by itself, and not as part of a [Bundle], an
+Envelope SHOULD use the suffix `.json`.
+
+-   For attestations intended for consumption by [in-toto-verify], an
+    Envelope containing an attestation about a particular SW supply chain
+    step `<step-name>` SHOULD be named `<step-name>.json`.
+-   For other verifiers, or cases in which a step name cannot be easily
+    determined, the attestation producer and consumer SHOULD agree on an
+    acceptable choice for filename: `<env-name>.json`.
+-   If multiple Envelopes are produced for the same step by different
+    [functionaries] uniquely identified by a public key, an Envelope name
+    SHOULD include the truncated [KEYID] of the public key `<keyid[0:8]>` of
+    the signing functionary: `<step/env-name>.<keyid[0:8]>.json`.
+
+## Storage convention
+
+The media type `application/vnd.in-toto.<predicate>+dsse` SHOULD
+be used to denote an individual attestation in arbitrary storage systems.
+
+-   The `<predicate>` MUST match the [predicate specification name]
+    without the file extension. Predicate versioning is handled in the
+    [Statement] layer.
+-   Consumers SHOULD NOT rely upon the media type for individual attestations
+    as faithful indicators of predicate type because this information is only
+    authenticated at the [Statement] layer.
+-   To obtain predicate information that is authenticated, consumers MUST
+    parse the Envelope's `payload`, and verify it against its `signatures`.
+
+### Examples
+
+Example media types for single DSSE-signed attestation predicates include:
```

Comment:
I think I finally see now. So these media types would be useful for _transporting_ an attestation requested on demand? It's a way for the server to say to clients, "Hey, this is the unathenticated media type for this attestation so you know in advance how to decode it, but please don't take my word for it---make sure you verify the signatures on the Envelope level first." Is that correct?

Reply:
Right, or conversely, it let's a client request attestations of a specific media type instead of having to get all of them.